### ISSUE TITLE
Hyperparameters and results

### ISSUE THREAD

**Issue by zbouchlih** [oooolga/IFT6390_Project#2]
### Linear regression

| Optimizer | epochs | learning rate | weight decay | test accuracy | validation accuracy | training accuracy | test loss | training loss | validation loss |dataset | 
|-----------|--------|---------------|--------------|---------------------|-------------------|---------------|-----------------|--------|-----------------|--------|
| SGD       |    100    |       0.05        |       0.0005       |       0.843     |  0.856        |        0.867           |  0.453  |  0.392         |   0.425              |  FMNIST     |
| Adam      |    100    |         0.05      |      0.0005        |   0.832    |  0.842          |       0.854          |   0.504   | 0.440       |     0.477          | FMNIST     |

### Polynomial regression
Total number of parameters: 31370

| Optimizer | epochs | learning rate | weight decay | test accuracy | validation accuracy | training accuracy | test loss | training loss | validation loss |dataset | 
|-----------|--------|---------------|--------------|---------------------|-------------------|---------------|-----------------|--------|-----------------|--------|
| SGD       |    100    |       0.01      |    0.0005        |      0.850      |    0.863      |         0.887          |  0.422  |    0.334       |        0.395         |       |
| Adam      |    100    |        0.01       |      0.0005        |   0.845    |     0.857        |        0.878          |   0.454   |   0.356      |     0.420            |      |

### Convolution Neural Network
Total number of parameters: 867658

| Optimizer | epochs | learning rate | weight decay | test accuracy | validation accuracy | training accuracy | test loss | training loss | validation loss |
|-----------|--------|---------------|--------------|---------------------|-------------------|---------------|-----------------|--------|-----------------|
| SGD   | 50 |  0.01 | 0.0005 | 0.931  | 0.934  | 0.993 | 0.220  | 0.206 |  0.029 |
| Adam |        |               |              |                     |                   |               |                 |      |   |

### Multi-layer Perceptron (MLP)
Total number of parameters: 456730

| Optimizer | epochs | learning rate | weight decay | test accuracy | validation accuracy | training accuracy | test loss | training loss | validation loss |
|-----------|--------|---------------|--------------|---------------------|-------------------|---------------|-----------------|--------|-----------------|
| SGD   | 50  |  0.01 | 0.0005  |    0.893 |  0.897  | 0.953  |  0.311   | 0.299 |  0.134 |
| Adam |        |               |              |                     |                   |               |                 |      |   |

### Linear regression

| Optimizer | epochs | learning rate | weight decay |test accuracy |  validation accuracy | training accuracy | training loss | validation loss |dataset | 
|-----------|--------|---------------|--------------|-------------------|---------------------|-------------------|---------------|-----------------|--------|
| SGD       |    100    |       0.001        |       0.0005       |       0.339         |            0.350        |       0.352          |       1.866        | 1.894                |  CIFAR10     |
| Adam      |     100    |      0.001        |      0.0005        |           0.348      |         0.330           |       0.338           |      1.884        |     01.904            | CIFAR10     |

### Polynomial regression
Total number of parameters: 122890

| Optimizer | epochs | learning rate | weight decay | test accuracy | validation accuracy | training accuracy | test loss | training loss | validation loss |dataset | 
|-----------|--------|---------------|--------------|---------------------|-------------------|---------------|-----------------|--------|-----------------|--------|
| SGD       |    100    |      0.01       |     0.0005       |            |          |                   |    |           |                 |       |
| Adam      |    100    |               |              |       |             |                  |      |         |                 |      |

### Convolution Neural Network
Total number of parameters:

| Optimizer | epochs | learning rate | weight decay | test accuracy | validation accuracy | training accuracy | test loss | training loss | validation loss |
|-----------|--------|---------------|--------------|---------------------|-------------------|---------------|-----------------|--------|-----------------|
| SGD   | 100 | 0.01 | 0.0005 |   0.870 | 0.867 | 0.991 |  0.055  |  0.433    | 0.443 |
| Adam |        |               |              |                     |                   |               |                 |      |  |

### Multi-layer Perceptron (MLP)
Total number of parameters: 5212338

| Optimizer | epochs | learning rate | weight decay | test accuracy | validation accuracy | training accuracy | test loss | training loss | validation loss |
|-----------|--------|---------------|--------------|---------------------|-------------------|---------------|-----------------|--------|-----------------|
| SGD   | 100  |  0.01 | 0.0005 |  0.572  |  0.596  |  0.657  |  1.231   | 1.161  | 0.955 |
| Adam |        |               |              |                     |                   |               |                 |      |   |

### Convolution Neural Network
Total number of parameters: 872431

| Optimizer | epochs | learning rate | weight decay | test accuracy | validation accuracy | training accuracy | test loss | training loss | validation loss |
|-----------|--------|---------------|--------------|---------------------|-------------------|---------------|-----------------|--------|-----------------|
| SGD   | 100 |  0.01  |   0.0005 | 0.897 |  0.890  | 0.943 | 0.303   |  0.293 |  0.143  |
| Adam |        |               |              |                     |                   |               |                 |      |    |

### Multi-layer Perceptron (MLP)
Total number of parameters: 460467

| Optimizer | epochs | learning rate | weight decay | test accuracy | validation accuracy | training accuracy | test loss | training loss | validation loss |
|-----------|--------|---------------|--------------|---------------------|-------------------|---------------|-----------------|--------|-----------------|
| SGD   | 100  |  0.01 | 0.0005 |  0.857  |  0.854  |  0.912  |  0.415   | 0.406 | 0.233 |
| Adam |        |               |              |                     |                   |               |                 |      |   |

### Polynomial regression
Total number of parameters: 147439

| Optimizer | epochs | learning rate | weight decay | test accuracy | validation accuracy | training accuracy | test loss | training loss | validation loss |dataset | 
|-----------|--------|---------------|--------------|---------------------|-------------------|---------------|-----------------|--------|-----------------|--------|
| SGD       |    100    |  0.01 |    0.0005        |     0.705       |   0.709       |    0.741               |   1.101 |    0.978       |      1.065           |       |
| Adam      |    100    |               |              |       |             |                  |      |         |                 |      |

### Linear regression

| Optimizer | epochs | learning rate | weight decay | test accuracy | validation accuracy | training accuracy | test loss | training loss | validation loss |dataset | 
|-----------|--------|---------------|--------------|---------------------|-------------------|---------------|-----------------|--------|-----------------|--------|
| SGD       |    100    | 0.05  |      0.0005      |       0.682     |      0.681    |      0.701            |1.180 |  1.112  |        1.142   |                 |      
| Adam      |    100    |        0.05       |        0.0005      |   0.627     |         0.628    |         0.644         |   1.344   |     1.269    |             1.303    |      |

**Comment by zbouchlih** [comment]
@Kangzf1996 if you can please fill this table while training the model and changing hyperparameters, feel free to add columns, it's just an example

**Comment by oooolga** [comment]
@zbouchlih @Kangzf1996 Hey, don't forget to record the evaluation results by running the eval.py file. Don't record result from the last epoch~~ This is very important!! :D
